GH-32739: [CI][Docs] Document Docs PR Preview (#35614)
apache/arrow: docs/source/developers/documentation.rst

   :ref:`docker-builds`.

> Pull Request Docs Preview
> -------------------------

Reviewer: ```suggestion
Building a docs preview in a Pull Request
-----------------------------------------
```

> You can use github-actions to build and host the documentation with crossbow within
> a pull request you are working on.

Reviewer: ```suggestion
You can build and preview the documentation within a GitHub pull request you are working on.
```

> To use github-actions add a comment ``@github-actions crossbow submit preview-docs``
> to the pull request. The rendered documentation will then be available within the
> github-actions response, where you need to click on the crossbow build badge:

Reviewer: ```suggestion
To do so, post the comment ``@github-actions crossbow submit preview-docs``
to the pull request. The rendered documentation will then be available within the
GitHub Actions response, where you need to click on the Crossbow build badge:
```
